edge/pkg/metamanager/metaserver/handlerfactory/handler.go

```diff
@@ -43,24 +45,33 @@ func NewFactory() Factory {
 		scope:             scope.NewRequestScope(),
 		MinRequestTimeout: 1800 * time.Second,
 		handlers:          make(map[string]http.Handler),
+		lock:              &sync.RWMutex{},
 	}
 	return f
 }
 
 func (f *Factory) Get() http.Handler {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
 	if h, ok := f.handlers["get"]; ok {
 		return h
 	}
 	h := handlers.GetResource(f.storage, f.storage, f.scope)
+	f.lock.Lock()
+	defer f.lock.Unlock()
 	f.handlers["get"] = h
 	return h
 }
 
 func (f *Factory) List() http.Handler {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
 	if h, ok := f.handlers["list"]; ok {
 		return h
 	}
 	h := handlers.ListResource(f.storage, f.storage, f.scope, false, f.MinRequestTimeout)
+	f.lock.Lock()
```

Comment:
deadlock

Reply:
@gccio 
ok, modified the lock type, and only locked once inside the function.
/ptal

Reply:
RWMutex lock is still used, and the lock is manually released to prevent deadlock caused by re-entry lock.